primitives/src/hash_to_group/mod.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2022 Espresso Systems (espressosys.com)
+// This file is part of the Jellyfish library.
+
+// You should have received a copy of the MIT License
+// along with the Jellyfish library. If not, see <https://mit-license.org/>.
+
+//! Hash to Elliptic Curve implementation of <https://datatracker.ietf.org/doc/draft-irtf-cfrg-hash-to-curve/>
+
+use ark_ec::{
+    short_weierstrass_jacobian::{GroupAffine, GroupProjective},
+    AffineCurve, SWModelParameters,
+};
+use ark_std::{
+    rand::{Rng, SeedableRng},
+    UniformRand,
+};
+use digest::Digest;
+use rand_chacha::ChaCha20Rng;
+use sha2::Sha256;
+
+use crate::errors::PrimitivesError;
+
+/// Trait definition and default implementation for hash to group functions.
+pub trait HashToGroup: SWModelParameters + Sized {
+    /// Hash to Group point, using sha2-512 function
+    /// hashing to G1 point of `C: ProjectiveCurve`.
+    // Default implementation implements a naive solution via rejection sampling.
+    // Slow, and non-constant time.
+    //
+    // For specific curves we may want to overload it with a more efficient
+    // algorithm, such as IETF BLS draft.
+    fn hash_to_group<B: AsRef<[u8]>>(
+        data: B,
+        cs_id: B,
+    ) -> Result<GroupProjective<Self>, PrimitivesError> {
+        let mut hasher = Sha256::new();
+        hasher.update([cs_id.as_ref(), data.as_ref()].concat());
+        let mut seed = [0u8; 32];
+        seed.copy_from_slice(hasher.finalize().as_ref());
+        let mut rng = ChaCha20Rng::from_seed(seed);
+        loop {
+            let x = Self::BaseField::rand(&mut rng);
+            let greatest = rng.gen();
+
+            if let Some(p) = GroupAffine::<Self>::get_point_from_x(x, greatest) {
+                return Ok(p.mul_by_cofactor_to_projective());
+            }
+        }
+    }
+}
+
+impl HashToGroup for ark_bls12_381::g1::Parameters {
+    // todo
```

Comment:
Minor: How about capitalizing `todo` to `TODO` so that it's easier to spot? (Same for other places.)